In [2]:
import pandas as pd
path="C:\\Users\\mf879\\OneDrive\\Desktop\\Karunadu\\Data\\Data\\Iris.csv"
data=pd.read_csv(path)
print(data)

      Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm  \
0      1            5.1           3.5            1.4           0.2   
1      2            4.9           3.0            1.4           0.2   
2      3            4.7           3.2            1.3           0.2   
3      4            4.6           3.1            1.5           0.2   
4      5            5.0           3.6            1.4           0.2   
..   ...            ...           ...            ...           ...   
145  146            6.7           3.0            5.2           2.3   
146  147            6.3           2.5            5.0           1.9   
147  148            6.5           3.0            5.2           2.0   
148  149            6.2           3.4            5.4           2.3   
149  150            5.9           3.0            5.1           1.8   

            Species  
0       Iris-setosa  
1       Iris-setosa  
2       Iris-setosa  
3       Iris-setosa  
4       Iris-setosa  
..              ...  
145  

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Id             150 non-null    int64  
 1   SepalLengthCm  150 non-null    float64
 2   SepalWidthCm   150 non-null    float64
 3   PetalLengthCm  150 non-null    float64
 4   PetalWidthCm   150 non-null    float64
 5   Species        150 non-null    object 
dtypes: float64(4), int64(1), object(1)
memory usage: 7.2+ KB


In [4]:
data.isnull().sum()

Id               0
SepalLengthCm    0
SepalWidthCm     0
PetalLengthCm    0
PetalWidthCm     0
Species          0
dtype: int64

In [11]:
inputs=data.drop(['Id','Species'],'columns')
outputs=data.drop(['Id','SepalLengthCm','SepalWidthCm','PetalLengthCm','PetalWidthCm'],'columns')

C:\Users\mf879\AppData\Local\Temp\ipykernel_10540\3993120199.py:1: FutureWarning: In a future version of pandas all arguments of DataFrame.drop except for the argument 'labels' will be keyword-only.
  inputs=data.drop(['Id','Species'],'columns')
C:\Users\mf879\AppData\Local\Temp\ipykernel_10540\3993120199.py:2: FutureWarning: In a future version of pandas all arguments of DataFrame.drop except for the argument 'labels' will be keyword-only.
  outputs=data.drop(['Id','SepalLengthCm','SepalWidthCm','PetalLengthCm','PetalWidthCm'],'columns')


In [12]:
inputs

SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm
0              5.1           3.5            1.4           0.2
1              4.9           3.0            1.4           0.2
2              4.7           3.2            1.3           0.2
3              4.6           3.1            1.5           0.2
4              5.0           3.6            1.4           0.2
..             ...           ...            ...           ...
145            6.7           3.0            5.2           2.3
146            6.3           2.5            5.0           1.9
147            6.5           3.0            5.2           2.0
148            6.2           3.4            5.4           2.3
149            5.9           3.0            5.1           1.8

[150 rows x 4 columns]

In [13]:
outputs

Species
0       Iris-setosa
1       Iris-setosa
2       Iris-setosa
3       Iris-setosa
4       Iris-setosa
..              ...
145  Iris-virginica
146  Iris-virginica
147  Iris-virginica
148  Iris-virginica
149  Iris-virginica

[150 rows x 1 columns]

In [17]:
import sklearn 
from sklearn.model_selection import train_test_split

In [18]:
x_train,x_test,y_train,y_test=train_test_split(inputs,outputs,test_size=0.2)

In [19]:
x_train

SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm
96             5.7           2.9            4.2           1.3
80             5.5           2.4            3.8           1.1
45             4.8           3.0            1.4           0.3
15             5.7           4.4            1.5           0.4
111            6.4           2.7            5.3           1.9
..             ...           ...            ...           ...
79             5.7           2.6            3.5           1.0
35             5.0           3.2            1.2           0.2
64             5.6           2.9            3.6           1.3
148            6.2           3.4            5.4           2.3
139            6.9           3.1            5.4           2.1

[120 rows x 4 columns]

In [20]:
x_test

SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm
106            4.9           2.5            4.5           1.7
16             5.4           3.9            1.3           0.4
141            6.9           3.1            5.1           2.3
146            6.3           2.5            5.0           1.9
143            6.8           3.2            5.9           2.3
113            5.7           2.5            5.0           2.0
115            6.4           3.2            5.3           2.3
7              5.0           3.4            1.5           0.2
55             5.7           2.8            4.5           1.3
110            6.5           3.2            5.1           2.0
136            6.3           3.4            5.6           2.4
10             5.4           3.7            1.5           0.2
123            6.3           2.7            4.9           1.8
17             5.1           3.5            1.4           0.3
92             5.8           2.6            4.0           1.2
68             6.2           2.2            4.5           1.5
89             5.5           2.5            4.0           1.3
36             5.5           3.5            1.3           0.2
97             6.2           2.9            4.3           1.3
23             5.1           3.3            1.7           0.5
41             4.5           2.3            1.3           0.3
9              4.9           3.1            1.5           0.1
125            7.2           3.2            6.0           1.8
81             5.5           2.4            3.7           1.0
20             5.4           3.4            1.7           0.2
58             6.6           2.9            4.6           1.3
82             5.8           2.7            3.9           1.2
11             4.8           3.4            1.6           0.2
126            6.2           2.8            4.8           1.8
5              5.4           3.9            1.7           0.4

In [21]:
y_train

Species
96   Iris-versicolor
80   Iris-versicolor
45       Iris-setosa
15       Iris-setosa
111   Iris-virginica
..               ...
79   Iris-versicolor
35       Iris-setosa
64   Iris-versicolor
148   Iris-virginica
139   Iris-virginica

[120 rows x 1 columns]

In [22]:
y_test

Species
106   Iris-virginica
16       Iris-setosa
141   Iris-virginica
146   Iris-virginica
143   Iris-virginica
113   Iris-virginica
115   Iris-virginica
7        Iris-setosa
55   Iris-versicolor
110   Iris-virginica
136   Iris-virginica
10       Iris-setosa
123   Iris-virginica
17       Iris-setosa
92   Iris-versicolor
68   Iris-versicolor
89   Iris-versicolor
36       Iris-setosa
97   Iris-versicolor
23       Iris-setosa
41       Iris-setosa
9        Iris-setosa
125   Iris-virginica
81   Iris-versicolor
20       Iris-setosa
58   Iris-versicolor
82   Iris-versicolor
11       Iris-setosa
126   Iris-virginica
5        Iris-setosa

In [25]:
from sklearn.neighbors import KNeighborsClassifier 


We will take K value is 13 because the formula to find k is 
> K=sqrt(No. of rows)\
> K has 150 rows root of that is 12.2 hence we will take 13

In [27]:
model=KNeighborsClassifier(n_neighbors=13)
model.fit(x_train,y_train)

C:\Users\mf879\AppData\Local\Programs\Python\Python310\lib\site-packages\sklearn\neighbors\_classification.py:228: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples,), for example using ravel().
  return self._fit(X, y)


KNeighborsClassifier(n_neighbors=13)

In [28]:
y_pred=model.predict(x_test)

In [29]:
print(y_pred)
print(y_test)

['Iris-versicolor' 'Iris-setosa' 'Iris-virginica' 'Iris-virginica'
 'Iris-virginica' 'Iris-virginica' 'Iris-virginica' 'Iris-setosa'
 'Iris-versicolor' 'Iris-virginica' 'Iris-virginica' 'Iris-setosa'
 'Iris-virginica' 'Iris-setosa' 'Iris-versicolor' 'Iris-versicolor'
 'Iris-versicolor' 'Iris-setosa' 'Iris-versicolor' 'Iris-setosa'
 'Iris-setosa' 'Iris-setosa' 'Iris-virginica' 'Iris-versicolor'
 'Iris-setosa' 'Iris-versicolor' 'Iris-versicolor' 'Iris-setosa'
 'Iris-versicolor' 'Iris-setosa']
             Species
106   Iris-virginica
16       Iris-setosa
141   Iris-virginica
146   Iris-virginica
143   Iris-virginica
113   Iris-virginica
115   Iris-virginica
7        Iris-setosa
55   Iris-versicolor
110   Iris-virginica
136   Iris-virginica
10       Iris-setosa
123   Iris-virginica
17       Iris-setosa
92   Iris-versicolor
68   Iris-versicolor
89   Iris-versicolor
36       Iris-setosa
97   Iris-versicolor
23       Iris-setosa
41       Iris-setosa
9        Iris-setosa
125   Iris-virginica


In [30]:
from sklearn.metrics import confusion_matrix
cm=confusion_matrix(y_pred,y_test)

In [31]:
print(cm)

[[11  0  0]
 [ 0  8  2]
 [ 0  0  9]]


In [32]:
11+8+9

28

In [34]:
print("Accuracy is ",(28/30)*100)

Accuracy is  93.33333333333333


In [35]:
res=model.predict([[4.5,2.3,4.5,1.2]])
print(res)

['Iris-versicolor']


C:\Users\mf879\AppData\Local\Programs\Python\Python310\lib\site-packages\sklearn\base.py:464: UserWarning: X does not have valid feature names, but KNeighborsClassifier was fitted with feature names
  warnings.warn(
